server/pulp/server/webservices/views/serializers/__init__.py

```diff
@@ -176,28 +176,78 @@ def to_representation(self, instance):
+        Recursively translate mongo queries from our external API to our internal db representation.
+
+        Mongo queries follow one of the following forms:
+            1. {"key": "value"} - ex. {"id": "thing1"}
+            2. {"key": {query}} - ex. {"id": {"$exists": 1}}
+            3. {"$key": [list of items] - ex. {'$in': ['this', 'or_this']
+            4. {"$key": [list of queries] - ex. {'$and': [{query_1}, {query_2}]
+            5. {"$key": {dict}} - ex. {"$elemMatch": {query}}
+
+        For cases 1 and 2, the key may need to be translated. Cases 3 - 5, the key does not need to
+        be translated because it is a mongo operator. For cases 2, 4, and 5, there may be nested
+        queries that will need to be translated. Possible nested queries are passed to the helper
+        function `_translate_nested_query` to recurse properly.
 
         :param model: the class that defines this document's fields
         :type  model: sublcass of mongoengine.Document
         :param filters: dict used to filter results
-        :type  filters: dict, following mongo syntax
+        :type  filters: dict, following mongo query syntax
 
-        :return: the same filters dictionary, with fields translated to the new db representation
-        :rtype:  dict, following mongo syntax
+        :return: filters dictionary, with fields translated
+        :rtype:  dict, following mongo query syntax
         """
         translated_dict = {}
         for key, value in filters.iteritems():
-            if key in self._remapped_fields.itervalues():
-                if key == '_id':
-                    translated_dict['_id'] = self._translate__id(value)
-                else:
-                    new_field = self.translate_field(model, key)
-                    translated_dict[new_field] = value
+            # Do not translate mongo operator keys, but value may contain queries (case 3-5)
+            if key.startswith('$'):
+                new_key = key
+                new_value = self._translate_nested_query(model, value)
             else:
-                translated_dict[key] = value
+                if key == '_id':  # _id is a special case
+                    new_key = '_id'
+                    new_value = self._translate__id(value)
+                else:
+                    # Case 1 and 2 keys may need to be translated
+                    remapped = self._remapped_fields.itervalues()
+                    new_key = self.translate_field(model, key) if key in remapped else key
+                    # Keep the value (case 1) or translate the nested query (case 2)
+                    new_value = self._translate_nested_query(model, value)
+            translated_dict[new_key] = new_value
         return translated_dict
 
+    def _translate_nested_query(self, model, query_piece):
+        """
+        Helper function to properly handle a nested piece of a query based on its type.
```

Comment:
nit: I only bring this up because I'm *super* guilty of doing this all the time in my docstrings. If it isn't helpful or doing things properly, why write it?

This should probably just be "Handle a nested piece of a query based on its type."